index.js

```diff
@@ -202,48 +218,36 @@ function buildArray (schema, code, name) {
 function nested (laterCode, name, key, schema) {
   var code = ''
   var funcName
-  if (schema.required) {
-    code += `
-      if (!obj.hasOwnProperty('${key.slice(1)}')) {
-        throw new Error('${key} is required!')
-      }`
-  }
   const type = schema.type
   switch (type) {
     case 'null':
       code += `
-        json += $asNull()
-      `
+        json += $asNull()`
       break
     case 'string':
       code += `
-        json += $asString(obj${key})
-      `
+        json += $asString(obj${key})`
       break
     case 'number':
     case 'integer':
       code += `
-        json += $asNumber(obj${key})
-      `
+        json += $asNumber(obj${key})`
       break
     case 'boolean':
       code += `
-        json += $asBoolean(obj${key})
-      `
+        json += $asBoolean(obj${key})`
       break
     case 'object':
       funcName = (name + key).replace(/[-.\[\]]/g, '')
       laterCode = buildObject(schema, laterCode, funcName)
       code += `
-        json += ${funcName}(obj${key})
-      `
+        json += ${funcName}(obj${key})`
       break
     case 'array':
       funcName = (name + key).replace(/[-.\[\]]/g, '')
       laterCode = buildArray(schema, laterCode, funcName)
       code += `
-        json += ${funcName}(obj${key})
-      `
+        json += ${funcName}(obj${key})`
```

Comment:
every generated code was put it in this way so that he could easily copied and pasted around, without worrying about backticks :). It's optimized to be read in the "template" form, rather than the generated one: I'm selfish and lazy :D.